File: Algorithms/Implementation/Drawing Book.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

class Result
{

    /*
     * Complete the 'pageCount' function below.
     *
     * The function is expected to return an INTEGER.
     * The function accepts following parameters:
     *  1. INTEGER n
     *  2. INTEGER p
     */

    public static int pageCount(int n, int p)
    {
        int totalPages = n / 2;
        int frontPages = p / 2;
        int backPages = totalPages - frontPages;
        return Math.Min(frontPages, backPages);
    }

}

class Solution
{
    public static void Main(string[] args)
    {
        TextWriter textWriter = new StreamWriter(@System.Environment.GetEnvironmentVariable("OUTPUT_PATH"), true);

        int n = Convert.ToInt32(Console.ReadLine().Trim());
        int p = Convert.ToInt32(Console.ReadLine().Trim());
        int result = Result.pageCount(n, p);

        textWriter.WriteLine(result);

        textWriter.Flush();
        textWriter.Close();
    }
}

File: Algorithms/Implementation/Drawing Book.php
<?php

/*
 * Complete the 'pageCount' function below.
 *
 * The function is expected to return an INTEGER.
 * The function accepts following parameters:
 *  1. INTEGER n
 *  2. INTEGER p
 */

function pageCount($n, $p) {
    # Write your code here
    $front = floor($p / 2);
    $back = floor(($n / 2) - floor($p / 2));
    return min($front, $back);
}

$fptr = fopen(getenv("OUTPUT_PATH"), "w");

$n = intval(trim(fgets(STDIN)));
$p = intval(trim(fgets(STDIN)));
$result = pageCount($n, $p);

fwrite($fptr, $result . "\n");
fclose($fptr);
